app/controllers/admin_controller.rb

```diff
@@ -123,6 +123,38 @@ def mark_spam
     end
   end
 
+  def mark_comment_spam
+    @comment = Comment.find params[:id]
+    if current_user && (current_user.role == 'moderator' || current_user.role == 'admin')
+      if @comment.status != 2
+        @comment.spam
+        flash[:notice] = "Comment has been marked as spam."
+      else
+        flash[:notice] = "Comment already marked as spam."
+      end
+    else
+      flash[:error] = 'Only moderators can moderate comments.'
+    end
+    redirect_to '/dashboard'
+  end
+
+  def publish_comment
+    if current_user && (current_user.role == 'moderator' || current_user.role == 'admin')
+      @comment = Comment.find params[:id]
+      if @comment.status != 2
```

Comment:
I thought you'd check to see if `status == 0`

Reply:
I wanted to but nodes and comments have a different status system.Active nodes has status = 1, spam has status 0.But, for comments, an active comment has status = 0 and so I have used status = 2(for spam).

Reply:
Oops, I understood your question now, actually, while writing tests I found that we have one more status for comment i.e., status = 1 - it represents what unknown to me, so I used condition 
status!= 2 so that other status values wouldn't change.

Reply:
I see your point.
Can status be set to a negative number?

Reply:
Got your point !!!